sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java

```diff
@@ -68,10 +71,31 @@ public BriteContentResolver wrapContentProvider(@NonNull ContentResolver content
   }
 
   /** An executable query. */
-  public interface Query {
+  public static abstract class Query {
     /** Execute the query on the underlying database and return the resulting cursor. */
     @CheckResult @WorkerThread
-    Cursor run();
+    public abstract Cursor run();
+
+    /**
+     * Execute the query on the underlying database and return an Observable of the rows queried.
+     *
+     * @param map Takes in a cursor for a single row and maps it to your desired result type.
+     */
+    public <T> Observable<T> map(final Func1<Cursor, T> map) {
```

Comment:
makes sense, map makes it sound like you're mapping the whole query to something. `asRows` sounds good.